packages/react-widgets/src/hooks/useWidgetFetch.js

```diff
@@ -15,20 +20,59 @@ export default function useWidgetFetch(
   const [data, setData] = useState();
   const [isLoading, setIsLoading] = useState(false);
   const [warning, setWarning] = useState('');
+  const dispatch = useDispatch();
 
-  const isSourceReady = useSelector(
-    (state) => global || selectAreFeaturesReadyForSource(state, dataSource)
-  );
   const source = useWidgetSource({ dataSource, id });
+  const { foreignSource, isForeignSourceReady, foreignFilterSelfColumn, foreignColumn } =
+    useSelector((state) => selectForeignFilterParams(state, source));
+  const hasForeignFilter = !!foreignSource;
+
+  const isSourceReady = useSelector((state) => {
+    const selfReady = global || selectAreFeaturesReadyForSource(state, dataSource);
+    if (hasForeignFilter) {
+      return selfReady && isForeignSourceReady;
+    }
+    return selfReady;
+  });
+
+  useEffect(() => {
```

Comment:
Great work here, this `useEffect` and the code relative to `isSourceReady` are the keys so that all the system works